framework/src/fvbcs/FVFluxBC.C

```diff
@@ -125,6 +125,22 @@ FVFluxBC::computeJacobian(const FaceInfo & fi)
 
   DualReal r = fi.faceArea() * fi.faceCoord() * computeQpResidual();
 
+#ifdef MOOSE_GLOBAL_AD_INDEXING
+  const auto & derivs = r.derivatives();
+
+  const auto & column_indices = derivs.nude_indices();
+  const auto & values = derivs.nude_data();
+
+  mooseAssert(column_indices.size() == values.size(), "Indices and values size must be the same");
+
+  const auto & row_indices = _var.dofIndices();
+  mooseAssert(row_indices.size() == 1, "We're currently built to use CONSTANT MONOMIALS");
+  const auto row_index = row_indices[0];
+
+  for (std::size_t i = 0; i < column_indices.size(); ++i)
+    _assembly.cacheJacobianContribution(row_index, column_indices[i], values[i], _matrix_tags);
+#else
```

Comment:
It seems like this sort of code will be very common and nearly identical for every system/object that ever will use global indexing for jacobians.  Maybe we need to wrap this up in some sort of global helper function or in a helper assembly function or something.  Then every system won't have to have these ifdefs everywhere - they can be consolidated into one place.

Reply:
I guess the challenge is that the global-indexing version will basically be identical for every system/object, but the local/current code for each system is somewhat specialized.  Maybe we have some sort of lambda func that is passed in to the call to the helper function - and the helper function calls the passed in function only if we aren't doing global indexing.

Reply:
I like that idea. Folks writing `computeJacobian` other than core MOOSE developers would find the lambda scary, but non-core developers shouldn't be overriding `computeJacobian` in general anyways